More improvements to Tileset.md (#46)
SuperTux/wiki: Tileset.md

1. [Introduction](#introduction)
2. [Tile Attributes](#tile-attributes)
   * [Attribute Combinations](#attribute-combinations)
   * [Deprecated Attributes](#deprecated-attributes)
3. [Tile Datas](#tile-datas)
   * [Slope Types](#slope-types)
4. [Adding Your Own Tiles](#adding-your-own-tiles)
   * [Initial Preparations](#step-1-initial-preparations)
   * [Adding Tiles](#step-2-adding-tiles)
   * [Adding Tilegroups](#step-3-adding-tilegroups)
5. [Special Tiles](#special-tiles)
   * [Adding Objects As Tiles](#adding-objects-as-tiles)
6. [Deprecated Tiles](#deprecated-tiles)
   * [Deprecating Your Old Tiles](#deprecating-your-old-tiles)

<hr>

Introduction

> ============

Reviewer: Under the "Introduction" -> "Examples" section, all separators are thicker. I believe they should be thinner as to not be mixed up with the ones separating the main parts of the document (1-6).

Author: Turns out there's no way of creating a thin line in MD without adding a title. I tried `<hr style="height:1px;">` but it doesn't seem to support CSS. I also tried to add a title with an invisible character, it made the line but created a really large empty space before the line, so, not sure on what to do about this

  (height 2)
  (ids
    3 4 5 6
    7 8 0 0)
  (attributes
    0 0 0 0
    1 1 0 0)
  (image "tiles/[tilegroup]/[multipletiles].png")
)
```

---

> For a tile that looks one way in editor and other way in game, you can set the images separately:

Reviewer: ```suggestion
For a tile which looks different in the editor, you can set the images separately:
```

```
(tiles
  (width 1)
  (height 1)
  (ids 9)
  (attributes 3)
  (images "tiles/[tilegroup]/[tile].png")
  (editor-images "tiles/[tilegroup]/[tile]-editor.png")
)
```

> If you want the tile to be invisible in game, you just remove the "(images)" section

Reviewer: ```suggestion
If you want the tile to be invisible in-game, remove the `(images)` section.
```

)
```

Special Tiles
=============
Some tiles are special, as they turn into objects (i.e: Bonus Blocks, Coins and Bricks) when the level is loaded.

Adding Objects As Tiles
-----------------------

Special tiles can only be defined one by one, that meaning, you can't turn a larger image into multiple tiles.
Turning a tile into an object is done by adding an `object-name` and an `object-data` within the `tile`.

The general structure for adding a special tile in your tileset is the following:
```
(tile
    (id [your tile ID])
    (images
      "objects/[object]/[object].png"
    )
    (object-name "[object class name]")
    (object-data "
      ([propriety1] 50)
      ([propriety2] \"[string]\")
    ")
)
```

> Be aware that any object propriety that has quotation marks on it, must get a backslash (\\) before any quotation marks, as they are quotation marks inside other quotation marks.

Reviewer: ```suggestion
Be aware that any quotation mark in `object-data` must be escaped by adding a backslash (`\`) before it, as it's a quotation mark inside other quotation marks.
```

---

> If you find it difficult to write the object proprieties you want to add as a tile, you can place this object in your level, open the level in a text editor and copy the information you'll find about this object, for example, if you want a Mr.Snowball, you get in the level file and search for it, you'll find:

Reviewer: ```suggestion
If you find it difficult to write the object proprieties you want to add as a tile, you can place the object in your level, open the level in a text editor and copy the properties present for the object. For example, if you add a Mr.Snowball to a level and customize it, in the level file you'll find:
```

```
(snowball
      (z-pos 52)
      (direction "left")
      (x 288)
      (y 128)
)
```

> From this information, you may extract what you want to turn into a tile, you must ignore the x and y positions as they will be defined as where the tile is.

Reviewer: ```suggestion
From this information, you may extract what you want to turn into a tile. You must ignore the `x` and `y` position properties as they will be defined, depending on where the tile is.
```

---
Let's turn that Mr. Snowball into a tile, then:

```
(tile
    (id 52)
    (images
      "creatures/snowball/snowball-0.png"
    )
    (object-name "snowball")
    (object-data "
      (z-pos 50)
      (direction \"left\")
    ")
)
```
---
For another example, here is a rock added as a tile:

```
(tile
    (id 53)
    (object-name "rock")
    (images
      "objects/rock/rock.png"
    )   
)
```
---
Now, a not-portable trampoline as a tile:

```
(tile
    (id 54)
    (images
      "objects/trampoline/trampoline2-0.png"
    )
    (object-name "trampoline")
    (object-data "
      (type \"stationary\")
      (z-pos 50)
    ")
)
```
---

> Now let's be creative, let's say you want to add a bonus block that is orange and has 2 Mr.Icecubes inside it:

Reviewer: ```suggestion
Now let's be creative, let's say you want to add a bonus block that is orange and has 2 Mr.Iceblocks inside it:
```

    )
    (object-name "snowball")
    (object-data "
      (z-pos 50)
      (direction \"left\")
    ")
)
```
---
For another example, here is a rock added as a tile:

```
(tile
    (id 53)
    (object-name "rock")
    (images
      "objects/rock/rock.png"
    )   
)
```
---
Now, a not-portable trampoline as a tile:

```
(tile
    (id 54)
    (images
      "objects/trampoline/trampoline2-0.png"
    )
    (object-name "trampoline")
    (object-data "
      (type \"stationary\")
      (z-pos 50)
    ")
)
```
---
Now let's be creative, let's say you want to add a bonus block that is orange and has 2 Mr.Iceblocks inside it:

```
(tile
    (id 55)
    (images
      "objects/bonus_block/orange-0.png"
    )
    (object-name "bonusblock")
    (object-data "
      (type \"orange\")
      (z-pos 51)
      (custom-contents
        (mriceblock
          (z-pos 50)
        )
      )
      (count 2)
      (contents \"custom\")
    ")
)
```
Again, if you feel confused with what information you must type in the object name and data sections, you can copy them from an object that's already placed in your level file!
<hr>

Deprecated Tiles
================

> When a tile becomes outdated due to it being replaced by a different tile / object / decal - or when it is no longer necessary for level design, this tile is usually made deprecated for the sake of compatibility with older levels that were built with them before. Deprecated tiles can still be loaded by the level and will work as they used to, but they're not available to build with, nor can they be copied when editing the level. 

Reviewer: ```suggestion
When a tile becomes outdated due to it being replaced by a different tile / object / decal, or when it is no longer necessary for level design, it is usually made deprecated for the sake of compatibility with older levels which include it. Deprecated tiles can still be loaded by the level and will work as they used to, but they cannot be added to tilegroups, nor can they be copied when editing the level.
```
